### ISSUE TITLE
Wrong spacing around c++ reference operator for an external function forward declared inside another function

### ISSUE THREAD

**Issue by mihaipopescu** [uncrustify/uncrustify#1251]
```diff
-    extern void BillboardRenderer_RenderMultiple(const RenderBatchedData& renderData, ShaderChannelMask channels);
+    extern void BillboardRenderer_RenderMultiple(const RenderBatchedData&renderData, ShaderChannelMask channels);
```

```cpp
void foo()
{
    extern void bar(const A& a);
}
```

Expected versus output diff:
---
```diff
diff --git "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60002-UNI-16283.cpp" "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60002-UNI-16283.cpp"
index 7f04264..a9f9987 100644
--- "a/C:\\work\\external\\uncrustify\\tests\\output\\staging\\60002-UNI-16283.cpp"
+++ "b/C:\\work\\external\\uncrustify\\tests\\results\\staging\\60002-UNI-16283.cpp"
@@ -1,5 +1,5 @@
 // It is deleting the space after the pointer marker
 void foo()
 {
-    extern void BillboardRenderer_RenderMultiple(const RenderBatchedData& renderData, ShaderChannelMask channels);
+    extern void BillboardRenderer_RenderMultiple(const RenderBatchedData&renderData, ShaderChannelMask channels);
 }
```

Debug support files:
---
[60002-UNI-16283.cpp](https://gist.githubusercontent.com/mihaipopescu/b19791101affb39ed1dac088d1745157/raw/7f042642b5ce3b0af953fe25891c1217f787eb48/60002-UNI-16283.cpp)
[60002-UNI-16283.cpp.log](https://gist.githubusercontent.com/mihaipopescu/b19791101affb39ed1dac088d1745157/raw/f6dc4cd0969bf1352b579b9a715c13561d42ca72/60002-UNI-16283.cpp.log)
[60002-UNI-16283.cpp.unc](https://gist.githubusercontent.com/mihaipopescu/b19791101affb39ed1dac088d1745157/raw/3fb99dd98fbe29ab60818467539efbbae9ce57da/60002-UNI-16283.cpp.unc)
[Uncrustify.Cpp.cfg](https://gist.githubusercontent.com/mihaipopescu/b19791101affb39ed1dac088d1745157/raw/23d62d9dba5caf352fdc1793792e98eed4b81929/Uncrustify.Cpp.cfg)
[UNI-16283.cpp](https://gist.githubusercontent.com/mihaipopescu/b19791101affb39ed1dac088d1745157/raw/7f042642b5ce3b0af953fe25891c1217f787eb48/UNI-16283.cpp)

Disclaimer:
---
Note: This is a generated issue from our internal issue tracker.
Expect a dry description which could be insufficient to properly understand the issue.
On the other hand, the support files should be sufficient to easily reproduce and debug the issue,
but please comment if it's still unclear and I will get back with clarifications.
See main discussion thread here #1223.

**Comment by CDanU** [comment]
Removal caused by 
```sql
# Add or remove space between type and word. Default=Force
sp_after_type                   = ignore   # ignore/add/remove/force
```
Also the ref char is moved with an empty config which should not happen.

**Comment by mihaipopescu** [comment]
> it should because `sp_after_type` defaults to `FORCE`

What do you mean by that? We have it on ignore and it should remain unchanged.

The removal might be caused by the `sp_addr=remove` but it should not happen if & it's part of a type according to description.

**Comment by CDanU** [comment]
> What do you mean by that? We have it on ignore and it should remain unchanged.

That was in reference to the first statement (that I striked). Because `sp_after_type` defaults to `FORCE` the ref char change, when an empty.cfg is used, is expected (sort of, I don't particularly like some of the defaults).

> but it should not happen
correct

**Comment by gmaurel** [comment]
Fixed at 1253321656d733365e269dca20d3b0bd592c11ce,  Dec 13 16:36:15 2017
